core/store/src/db.rs

```diff
@@ -202,8 +202,7 @@ impl Database for RocksDB {
             let cf_handle = &*self.cfs[col as usize];
             let iterator = self
                 .db
-                .iterator_cf_opt(cf_handle, &self.read_options, IteratorMode::Start)
-                .unwrap();
+                .iterator_cf_opt(cf_handle, rocksdb_read_options(), IteratorMode::Start);
```

Comment:
I reviewed RocksDB implementation and it seems that they opted to require a new ReadOptions instance on every iterator to avoid dealing with lifetimes :man_shrugging:

`get_cf_opt` still uses the precomputed `self.read_options` value.

Reply:
Yes